### PR TITLE
SNOW-878562 - Release 1.6.0

### PR DESCRIPTION
**Pull request by sfc-gh-anavalos**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-aalam — COMMENTED** [review]
Do we do changelog review as well in this PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.6.0 (TBD)
+## 1.6.0 (2023-07-28)
```
Comment (sfc-gh-sfan) [inline CHANGELOG.md:3]: Line 14 seems to have a wrong indentation

**Review by sfc-gh-sfan — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-yixie) [reply CHANGELOG.md:3]: Are we waiting for the connector to be released first?

Comment (sfc-gh-aalam) [reply CHANGELOG.md:3]: > Line 14 seems to have a wrong indentation

It is a sub-bullet point for the auto-upload feature

Comment (sfc-gh-sfan) [reply CHANGELOG.md:3]: Oh okay 🤦 I guess I need a break lmao

**Comment by sfc-gh-stan** [comment]
I'm hoping to merge https://github.com/snowflakedb/snowpark-python/pull/964 before the release.

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: You might need to update the date from 28 to 31.

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: I don't know why, but I can't add feedback to specific lines below this point. Maybe the content was added in a different PR or something? Anyway, here is my feedback about the rest of the new changelog content:

Change: 
```
- Added support for new functions in `snowflake.snowpark.functions`:
```
to 
```
"- Added support for these new functions in `snowflake.snowpark.functions`:"
```

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: Change 
```
- Added support for pure Python packages specified via `Session.add_requirements` or `Session.add_packages`, now usable in stored procedures and UDFs even if packages are not present on the Snowflake Anaconda channel.
```

to

```
- Added support for pure Python packages specified via `Session.add_requirements` or `Session.add_packages`. They are now usable in stored procedures and UDFs even if packages are not present on the Snowflake Anaconda channel.
```

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: Change

```
- Added support for async execution of multi-query dataframe containing binding variables.
```

to

```
- Added support for asynchronous execution of multi-query dataframes that contain binding variables.
```

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: Regarding this line, I'm not sure you should mention private preview features. If you do decide to include it, please spell out what UDAF stands for like this "user-defined aggregate functions (UDAFs) "

```
- Added support for UDAF. This feature is currently in private preview.
```

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: SQL should be capitalized in this sentence:

```
- Fixed a bug where `DataFrame.union` was not generating the correct `schema_query` when sql simplifier is enabled.
```

**Review by sfc-gh-mabrennan — COMMENTED** [review]
I added some comments about the changelog entries. Thanks.

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-stan) [reply CHANGELOG.md:3]: Hi Mary, I'm addressing these comments, for the last one, I don't think `schema_query` is a SQL term here. cc: @sfc-gh-aalam

Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:3]: @sfc-gh-stan Sorry, I think I wasn't clear in my comment. I'm not suggesting that you capitalize "schema_query". I'm just saying that when we talk about the SQL language in general, the name of the language, SQL, should be capitalized. So the sentence would be ..."...when SQL simplifier...".

Comment (sfc-gh-stan) [reply CHANGELOG.md:3]: Gotcha, thanks Mary!

**Comment by sfc-gh-stan** [comment]
> I'm hoping to merge #964 before the release.

#964 is merged and I have rebased this branch.

**Pull request merged**